eng/Versions.props

```diff
@@ -25,7 +25,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <!-- Versions used by several individual references below -->
-    <RoslynDiagnosticsNugetPackageVersion>2.9.3</RoslynDiagnosticsNugetPackageVersion>
+    <RoslynDiagnosticsNugetPackageVersion>2.9.4</RoslynDiagnosticsNugetPackageVersion>
```

Comment:
@mavasni Was this the buggy version that had that big memory leak?

Reply:
Memory leak has been there since 2.9.3, but we explicitly turned off all the DFA rules that could cause it in our solution ruleset. Not sure if 2.9.4 has more DFA rules on by default. I am planning to publish 2.9.5-beta1 which has the fix for the memory leak, and will submit a follow-up PR to move to that package before we merge this branch back to master.